### PR TITLE
Better CMake support

### PR DESCRIPTION
**Pull request by knoepfel**
The changes here allow building `root_serialization` in (non-)UPS environments using homogeneous CMake interface.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
@knoepfel should the README file be updated to reflect any new build arguments?

File: pds_reading.cc [hunk 1]
```diff
@@ -29,7 +29,7 @@ namespace {
 std::pair<uint32_t, Compression> readPreamble(std::istream& iFile) {
   std::array<uint32_t, 4> header;
   iFile.read(reinterpret_cast<char*>(header.data()),4*4);
-  assert(file_.rdstate() == std::ios_base::goodbit);
+  assert(iFile.rdstate() == std::ios_base::goodbit);
```
Comment (Dr15Jones) [inline pds_reading.cc:32]: nice catch! Did the debug build fail?

**Comment by Dr15Jones** [comment]
@gartung since I am a CMake neophyte (at best), could you take a quick look at the CMake files?

File: pds_reading.cc (hunk 1 above)
Comment (knoepfel) [reply pds_reading.cc:32]: Yep--only way I would have found it. :)

**Comment by knoepfel** [comment]
@Dr15Jones, yes I should update the README file.

**Comment by Dr15Jones** [comment]
@knoepfel will you be updating the README as part of this pull request or do you want to do it in another one?

**Comment by knoepfel** [comment]
I'll include the updated README as part of the PR later today.

**Comment by gartung** [comment]
@Dr15Jones This should still work with the cmake command line used with my pull request
```
cmake .. -DZSTD_DIR=/cvmfs/cms.cern.ch/slc7_amd64_gcc900/external/zstd/1.4.5/lib/cmake -DTBB_DIR=/cvmfs/cms.cern.ch/slc7_amd64_gcc900/external/tbb/2020_U3/cmake -DLZ4_DIR=/cvmfs/cms.cern.ch/slc7_amd64_gcc900/external/lz4/1.9.2/  -DROOT_DIR=/uscms_data/d1/cdj/cce_measurements/new_root/install/cmake
```

**Review by gartung — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -8,14 +8,22 @@ The CMakeLists.txt uses 3rd party packages from /cvmfs which are identical to th
...
-1. cmake .. -DZSTD_DIR=path_to_zstd_install -DTBB_DIR=path_to_tbb_install -DLZ4_DIR=path_to_lz4_install -DROOT_DIR=path_to_root_install
-1. make
 
-This will create the executable `threaded_io_test`.
+```console
+$ git clone <root_serialization URL>
+$ mkdir build-root-serialization
+$ cd build-root-serialization
+$ cmake <path to root_serialization> \
+  -DROOT_DIR=$ROOTSYS/cmake \
```
Comment (Dr15Jones) [inline README.md:17]: This assumes ROOT's runtime has already been setup. That hasn't been the case when I do the build.

Comment (knoepfel) [reply README.md]: Adjusted.

**Review by Dr15Jones — APPROVED** [review]

**Pull request merged**